script/scaffold/templates/config_flow/tests/test_config_flow.py

```diff
@@ -8,7 +11,16 @@
 from homeassistant.data_entry_flow import FlowResultType
 
 
-async def test_form(hass: HomeAssistant) -> None:
+@pytest.fixture(autouse=True, name="mock_setup_entry")
+def override_async_setup_entry() -> Generator[AsyncMock, None, None]:
```

Comment:
Not sure why the name of the fixture is set, we also don't use the `override_*` naming generally... so that looks odd to me.

Why not just name the function `mock_setup_entry` ?

Additionally, I think it is better practice (and better readable) if these things are in `conftest.py` and not in top of this file (it distracts from the actual tests in this module).

../Frenck

Reply:
I have opened #88481 to rename the function.

Reply:
I dunno, this will now result in a review comment 9 out of 10 times asking to move it, which is also not great 🤷

Reply:
I have opened an alternative PR #88484

Reply:
For reference, name is set so function def can differ from fixture name. If fixture name is in same as function name, you get a linting warning on the test cases that a testcase parameter hides a function.

Same issue will occur if a fixture uses another fixter inside conftest.

Reply:
> For reference, name is set so function def can differ from fixture name.

I'm aware, although, it isn't an issue and we have explicitly disabled those linting checks. Nevertheless, in general we use the `_fixture` postfix to resolve such cases.